server/src/main/java/io/pravega/schemaregistry/service/SchemaRegistryService.java

```diff
@@ -864,6 +873,29 @@ private SchemaInfo normalizeSchemaBinary(SchemaInfo schemaInfo) {
         return new SchemaInfo(schemaInfo.getType(), schemaInfo.getSerializationFormat(), schemaBinary, schemaInfo.getProperties());
     }
 
+    private void validateJsonSchema(String schemaString) {
+        try {
+            // 1. try draft 3
+            // jackson JsonSchema only supports json draft 3. If the schema definition is not compatible with draft 3, 
+            // try parsing the schema with everit library which supports drafts 4 6 and 7. 
+            OBJECT_MAPPER.readValue(schemaString, JsonSchema.class);
+        } catch (IOException e) {
```

Comment:
Is `IOException` in this case is always indicative of JSON schema version problem? 

Or, should the `JsonParseException` and `JsonMappingException` exceptions (both child types of `IOException` too) be caught here instead? In your tests, if the two exceptions are throws due to the difference in JSON schema version, perhaps we should catch these exceptions instead. 

What d you say?

Reply:
we can make it JsonParseException.